src/cargo/util/toml/schema.rs

```diff
@@ -670,6 +670,98 @@ pub struct TomlProfile {
     pub trim_paths: Option<TomlTrimPaths>,
 }
 
+impl TomlProfile {
+    /// Overwrite self's values with the given profile.
+    pub fn merge(&mut self, profile: &Self) {
```

Comment:
> Profile layering was one of the few pieces of logic I kept in the schema, assuming its general enough / decoupled enough from cargo policies.

How is thing general enough? Do we have any possible use case outside cargo?

Reply:
From what I understand of profiles, the merging is inherent to their definition, so if someone wants to do process the manifest, they'll need it.

I know its a bit dubious about why this when other things don't get included.  The biggest motivation is that this is tightly coupled to the struct definition and keeping them close is important to avoid forgetting to merge new fields.  The reason I felt this as justifiable was that it is very direct without special logic.